compiler/rustc_mir_build/src/build/expr/stmt.rs

```diff
@@ -40,7 +40,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 // Generate better code for things that don't need to be
                 // dropped.
                 if lhs.ty.needs_drop(this.tcx, this.param_env) {
-                    let rhs = unpack!(block = this.as_local_operand(block, rhs));
+                    let rhs = unpack!(block = this.as_local_rvalue(block, rhs));
```

Comment:
It's a little surprising to me that this wasn't `.as_local_rvalue` already. Were we just less precise than we could be before?

Reply:
Oh no, I guess this change is because `build_drop_and_replace` was changed to take an rvalue.